diff -NurpP --minimal libpng-1.5.7/config.sub libpng-1.5.7-xenon/config.sub
--- libpng-1.5.7/config.sub	2011-11-24 19:08:12.000000000 +0100
+++ libpng-1.5.7-xenon/config.sub	2012-01-28 23:21:20.043131805 +0100
@@ -814,6 +814,14 @@ case $basic_machine in
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
+	xenon)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
