diff -NurpP --minimal freetype-2.4.8/builds/unix/config.sub freetype-2.4.8-xenon/builds/unix/config.sub
--- freetype-2.4.8/builds/unix/config.sub	2011-11-15 07:53:50.000000000 +0100
+++ freetype-2.4.8-xenon/builds/unix/config.sub	2012-01-28 23:09:39.071125889 +0100
@@ -1001,6 +1001,14 @@ case $basic_machine in
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
+	xenon)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
