diff -aur newlib-1.20.0/config.sub newlib-1.20.0/config.sub
--- newlib-1.20.0/config.sub	2008-12-18 04:26:46.000000000 +0100
+++ newlib-1.20.0/config.sub	2009-04-12 20:36:55.000000000 +0200
@@ -919,6 +919,14 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
+	xenon)
+		basic_machine=powerpc64-unknown
+		os=-none
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
